src/database/hooks/useLocalDatabaseHook.js

```diff
@@ -10,14 +10,15 @@ import {InitialStartupAtom} from '../../service/initialStartup';
 
 const MIGRATION_STATUS = {
   NOT_MIGRATED: 'NOT_MIGRATED',
-  MIGRATED: 'MIGRATED'
+  MIGRATED: 'MIGRATED',
+  MIGRATING: 'MIGRATING'
 };
 
 /**
  *
  * @returns {UseLocalDatabaseHook}
  */
-const useLocalDatabaseHook = () => {
+const useLocalDatabaseHook = (withMigration = false) => {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The addition of the `withMigration` parameter to the `useLocalDatabaseHook` function is a good change. It allows for conditional migration, which can be useful in scenarios where you might not want to migrate the database immediately or under certain conditions.
```diff
- const useLocalDatabaseHook = () => {
+ const useLocalDatabaseHook = (withMigration = false) => {
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->